BookingsAPI/Bookings.DAL/Migrations/20190708130456_AddQuestionnaireTable.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Bookings.DAL.Migrations
+{
+    public partial class AddQuestionnaireTable : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropForeignKey(
+                name: "FK_SuitabilityAnswer_Participant_ParticipantId",
+                table: "SuitabilityAnswer");
+
+            migrationBuilder.DropIndex(
+                name: "IX_SuitabilityAnswer_ParticipantId",
+                table: "SuitabilityAnswer");
+
+            migrationBuilder.DropColumn(
+                name: "ParticipantId",
+                table: "SuitabilityAnswer");
+
+            migrationBuilder.AddColumn<long>(
+                name: "QuestionnaireId",
+                table: "SuitabilityAnswer",
+                nullable: false,
+                defaultValue: 0L);
+
+            migrationBuilder.AddColumn<long>(
+                name: "QuestionnaireId",
+                table: "Participant",
+                nullable: true);
+
+            migrationBuilder.CreateTable(
+                name: "Questionnaire",
+                columns: table => new
+                {
+                    Id = table.Column<long>(nullable: false)
+                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
+                    ParticipantId = table.Column<Guid>(nullable: false),
+                    UpdatedDate = table.Column<DateTime>(nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_Questionnaire", x => x.Id);
+                    table.ForeignKey(
+                        name: "FK_Questionnaire_Participant_ParticipantId",
+                        column: x => x.ParticipantId,
+                        principalTable: "Participant",
+                        principalColumn: "Id",
+                        onDelete: ReferentialAction.Cascade);
```

Comment:
I don't know if that is the way it is but it kind of looks like this cascade is going the other way so that if you remove the questionnaire, the participant would be deleted?

Reply:
EF will delete Questionnaire record and set it's Id to null in the parent record.
If the parent record deleted then Questionnaire record deleted.